app/backend/src/couchers/models.py

```diff
@@ -176,6 +182,17 @@ def mutual_friends(self, target_id):
 
         return session.query(User).filter(User.id.in_(q1.union(q2).intersect(q3.union(q4)).subquery())).all()
 
+    def sync_jail(self):
+        """
+        Sync all sessions with current jail status.
+
+        The auth interceptor does not check users table to reduce per-request DB overhead and this info is cached in the user session
```

Comment:
Add more indices to the database instead if you worry about performance. No duplicated state please!